engine/src/main/java/org/terasology/world/block/items/BlockItemSystem.java

```diff
@@ -112,6 +121,28 @@ public void onPlaceBlock(ActivateEvent event, EntityRef item) {
         }
     }
 
+    /**
+     * Returns the position at which the block side was hit, relative to the side.
+     * <p/>
+     * Example: The front side was hit right in the center.
+     * The result will be (0.5, 0.5), representing the relative hit position on the side's surface.
+     * @param hitPosition the hit position
+     * @param blockPosition the block position relative to its center (block (0, 0, 0) has block position (0.5, 0.5, 0.5))
+     * @return the 2D hit position relative to the side that was hit
+     */
+    private Vector2f getSideHitPosition(Vector3f hitPosition, Vector3f blockPosition) {
+        float epsilon = 0.0001f;
+        Vector3f relativeHitPosition = new Vector3f(hitPosition).sub(blockPosition);
+
+        if (Math.abs(relativeHitPosition.x) > 0.5f - epsilon) {
```

Comment:
I'm wondering whether we have convenience for that in TeraMath....

Reply:
I don't know, feels rather specific.
You mean a convenience for "testing if the absolute of a value is bigger than some value, considering some delta"?
There are no utility functions in TeraMath using the absolute functions, so I'm pretty sure it's not in there.
Also, I'm not sure how to name this, in case I wanted to move it to TeraMath. Any ideas?

Reply:
Not necessarily with the `Math.abs` but the comparison of floating point numbers with some ε